### ISSUE TITLE
Model will be uploaded twice with same storage path

### ISSUE THREAD

**Issue by Flippchen** [clearml/clearml#1395]
## Describe the bug
When training models and uploading them using `OutputModel`, the model is uploaded **twice** with different model IDs but to the **same storage location**.

Only one of these models is listed in the task output, but both models are created by the same task.

Even when explicitly setting:

```python
auto_connect_frameworks={'pytorch': False}
```

…the model still gets uploaded twice.

#### Suspected Side Effect

I believe this also leads to a secondary issue:

When deleting 

![Image](https://github.com/user-attachments/assets/656685df-f9cb-4bd7-b53f-32503af7a6ec)
![Image](https://github.com/user-attachments/assets/73320219-6967-4892-bc9a-4e2934906a1a)
![Image](https://github.com/user-attachments/assets/20379ec3-2370-4483-9f15-8ce81591c4e7)

a task (but keeping artifacts that are still in use), one of the two model entries may still be used (e.g. for production or inference), while the second model is removed.

As a result:
- The file is deleted from storage.
- The remaining model entry still appears valid in the UI.
- However, trying to access the model weights fails because the actual file no longer exists.

## To reproduce
0. Set `autoconnect` for PyTorch to `False`.
1. Train a YOLO model. (A colleque tested it also with e.g keras)
2. Create an `OutputModel`:
    ```python
    run_path = os.path.join(f'runs/segment/{training_instance.save_name}')

    output_model_best = OutputModel(task=task)
    output_model_best.update_labels(labels)
    output_model_best.update_weights(weights_filename=f"{run_path}/weights/best.pt")
    output_model_best.name = training_instance.save_name
    output_model_best.tags = project_params["model_tags"]
    ```
3. Train model (potentially on a remote agent using `execute_remotely`; **not tested locally**).
4. Check the task's **Output** tab → Only one model is listed (`OutputModel`).
5. Navigate to the **Models** tab → **Two models** are listed:
    - One with labels, metadata, etc.
    - One without any labels or metadata.
    - Both models have **different model IDs**, but share the **same storage path**.
6. Cleanup bug:
    - Archive the task.
    - Use the labeled model in another (unarchived) task.
    - Run the cleanup service.
    - One of the models is deleted, including the shared storage file.
    - In the UI, the model still appears valid, but downloading it results in a **404 Not Found** error.

## Expected behaviour
- Only one model instance should be created with a unique storage path.
- Cleanup should not delete shared storage if still in use by other tasks/models.
- The UI should reflect the actual availability of models correctly.

## Environment
* Type: self hosted
* ClearML SDK Version: '1.16.5'
* WebApp: 1.16.2-502 • Server: 1.16.2-502 • API: 2.30
* Python Version: 3.10
* OS (Windows \ Linux \ Macos): Linux and Windows

**Comment by eugen-ajechiloae-clearml** [comment]
Hi @Flippchen ! Thank you for reporting the issue and providing a detailed description. We have added it to our roadmap.
